lib/src/migrators/division.dart

```diff
@@ -482,3 +488,41 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
     _inCalcContext = previousCalcContext;
   }
 }
+
+/// Represents whether an expression is definitely a number, definitely not a
+/// number, or unknown.
+enum _NumberStatus {
+  yes,
+  no,
+  maybe;
+
+  /// Returns [yes] if [node] is definitely a number, [no] if [node] is
+  /// definitely not a number, and [maybe] otherwise.
+  static _NumberStatus of(Expression node) {
+    switch (node) {
```

Comment:
Another good candidate for expression form

Reply:
Done